contracts/gateway/EIP20Gateway.sol

```diff
@@ -1159,8 +1151,13 @@ contract EIP20Gateway is GatewayBase {
         );
 
         unstakeAmount_ = redeemAmount_.sub(rewardAmount_);
+
+        address beneficiary = unstakes[_messageHash].beneficiary;
+
+        delete unstakes[_messageHash];
```

Comment:
This delete does not prevent the execution of the next line `stakeVault.releaseTo(beneficiary, unstakeAmount_);`...